core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -274,7 +274,9 @@ private[spark] class TaskSchedulerImpl(
       maxLocality: TaskLocality,
       shuffledOffers: Seq[WorkerOffer],
       availableCpus: Array[Int],
-      tasks: IndexedSeq[ArrayBuffer[TaskDescription]]) : Boolean = {
+      tasks: IndexedSeq[ArrayBuffer[TaskDescription]],
+      addresses: ArrayBuffer[String],
+      taskDescs: ArrayBuffer[TaskDescription]) : Boolean = {
```

Comment:
instead of filling 2 arrays here, can we just fill one `ArrayBuffer[(String, Int)]`?

Reply:
Will this help save some memory or something? I was wondering what shall be the name of the combined ArrayBuffer but couldn't figure out a perfect one.

Reply:
it's just cleaner. how about `addressesWithIndice`